pkg/collector/autodiscovery/templatecache.go

```diff
@@ -18,6 +18,14 @@ type TemplateCache struct {
 	digest2ids      map[string][]string     // map a config digest to the list of AD identifiers it has
 	digest2template map[string]check.Config // map a digest to the corresponding config object
 	m               sync.RWMutex
+	CPupdate        map[string]CPAdIds
```

Comment:
I'd rather this moves to a util class used by the 3 KV providers, instead of adding more complexiting in the TemplateCache & the AutoConfig. That should be a helper class similar to tagger.utils.Expire or kubelet.PodWatcher that providers can keep as a private class member.